sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2224,6 +2253,17 @@ object EliminateSubqueryAliases extends Rule[LogicalPlan] {
 }
 
 /**
+ * Removes [[View]] operators from the plan. The operator is only used to provide the database
+ * name we should use in resolving a view, and it's respected till the end of analysis stage
+ * just because we want to see which part of a analyzed logical plan is generated from a view.
+ */
+object EliminateView extends Rule[LogicalPlan] {
```

Comment:
This rule is only used by `Optimizer`, it's placed here because it is in fact a analyze rule.